<!-- MTracker with automatic controller -->
<launch>
  <env name="ROS_MASTER_URI" value="http://MTracker:11311"/>
  <env name="ROSLAUNCH_SSH_UNKNOWN" value="1"/>

  <machine name="MTracker" user="mtracker" address="MTracker" env-loader="/home/mtracker/catkin_ws/src/mtracker/scripts/mtracker_env.sh" default="true"/>
  <machine name="MTrackerUI" user="tysik" address="MTrackerUI" env-loader="/home/tysik/workspace/catkin_ws/src/mtracker/scripts/mtracker_ui_env.sh"/>

  <param name="loop_rate" value="100"/>
  <param name="controls_topic" type="string" value="controls"/>
  <param name="joy_topic" type="string" value="joy"/>
  <param name="keys_topic" type="string" value="keys"/>
  <param name="odom_pose_topic" type="string" value="odom_pose"/>
  <param name="odom_velocity_topic" type="string" value="odom_velocity"/>

  <!-- Mtracker Nodes -->
  <node name="manual_controller" pkg="mtracker" type="manual_controller" respawn="false" required="true" output="screen" machine="MTracker">
    <param name="linear_gain" value="0.4"/>
    <param name="angular_gain" value="1.5"/>
  </node>

  <node name="reference_generator" pkg="mtracker" type="reference_generator" respawn="false" required="true" output="screen" machine="MTracker">
    <param name="trajectory_type" value="2"/>
    <param name="trajectory_paused" value="false"/>
    <param name="parent_frame" type="string" value="world"/>
    <param name="child_frame" type="string" value="reference"/>
  </node>

  <node name="mtracker" pkg="mtracker" type="mtracker" respawn="false" required="true" output="screen" machine="MTracker">
    <!-- Here put some parameters... -->
  </node>


  <!-- PC Nodes -->
  <node name="joy" pkg="joy" type="joy_node" respawn="false" required="true" output="screen" machine="MTrackerUI">
    <param name="dev" type="string" value="/dev/input/js0"/>
    <param name="deadzone" value="0.0"/>
    <param name="autorepeat_rate" value="100"/>
  </node>

  <node name="rviz" pkg="rviz" type="rviz" respawn="false" required="true" output="screen" machine="MTrackerUI" args="-d $(find mtracker)/resources/mtracker.rviz"/>
</launch>

